optuna/integration/pytorch_lightning.py

```diff
@@ -29,18 +29,50 @@ class PyTorchLightningPruningCallback(EarlyStopping):
             ``pytorch_lightning.LightningModule.training_step`` or
             ``pytorch_lightning.LightningModule.validation_epoch_end`` and the names thus depend on
             how this dictionary is formatted.
+        min_delta: minimum change in the monitored quantity
+            to qualify as an improvement, i.e. an absolute
+            change of less than ``min_delta``, will count as no
+            improvement. Default: ``0.0``.
+        patience: number of validation epochs with no improvement
+            after which training will be stopped. Default: ``3``.
+        verbose: verbosity mode. Default: ``False``.
+        mode: one of {``min``, ``max``}. In ``min`` mode,
+            training will stop when the quantity
+            monitored has stopped decreasing; in ``max``
+            mode it will stop when the quantity
+            monitored has stopped increasing. Default: ``min``.
+        strict: whether to crash the training if ``monitor`` is
+            not found in the validation metrics. Default: ``True``.
```

Comment:
In the original [`pytorch_lightning.callbacks.EarlyStopping`](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/callbacks/early_stopping.py), the stopping rule at the end of validation is Pytorch Lightning's own. However, in this `optuna.integration.PyTorchLightningPruningCallback`, the stopping rule is Optuna's own, which is specified by the pruner object in the `optuna.pruners` module. The stopping rule is implemented in `EarlyStopping.on_validation_end`, but it is overwritten in `PyTorchLightningPruningCallback`.

On the other hand, the stopping rule at the end of epochs is implemented in `EarlyStopping` but isn't in `PytorchLightningPruningCallback`. So, the same rule is applied to both of them. However, IMO, the `PytorchLightningPruningCallback` is designed to use Optuna's pruning feature in PyTorch Lightning, so we should not use it for other purposes. I think if we want to perform the early stopping at the end of epochs, we should define another callback as follows.
```python
callbacks = [
    PyTorchLightningPruningCallback(trial, monitor="val_acc"),
    pl.callbacks.EarlyStopping(monitor="val_acc", mode="max"),
]
```

Therefore, I think we should not add these parameters to `PyTorchLightningPruningCallback`.